src/sst/core/elemLoader.cc

```diff
@@ -135,92 +73,114 @@ splitPath(const std::string& searchPaths)
     return paths;
 }
 
-static void
-followError(
-    const std::string& libname, const std::string& elemlib, const std::string& searchPaths, std::ostream& err_os)
+ElemLoader::ElemLoader(const std::string& searchPaths) : searchPaths(searchPaths)
 {
-    std::string so_path = libname + ".so";
-    std::string fullpath;
-    void*       handle;
-
-    std::vector<std::string> paths = splitPath(searchPaths);
+    preload_symbols();
+    verbose = false;
 
-    for ( std::string path : paths ) {
-        struct stat sbuf;
-        int         ret;
+    const char* verbose_env = getenv("SST_CORE_DL_VERBOSE");
+    if ( nullptr != verbose_env ) { verbose = atoi(verbose_env) > 0; }
 
-        fullpath = path + "/" + so_path;
-        ret      = stat(fullpath.c_str(), &sbuf);
-        if ( ret == 0 ) break;
+    const char* bind_env = getenv("SST_CORE_DL_BIND_POLICY");
+    if ( nullptr == bind_env ) { bindPolicy = RTLD_LAZY | RTLD_GLOBAL; }
```

Comment:
Can skip this if and be more clear if you just construct `bindPolicy = RTLD_LAZY | RTLD_GLOBAL;` Then you only have the cases that `bind_env != nullptr` to contend with.

Reply:
So much for self-documenting code.

Reply:
? Why would you not want to initialize it to the expected default value.